lib/oli/delivery/settings.ex

```diff
@@ -29,6 +31,27 @@ defmodule Oli.Delivery.Settings do
     combine(resolved_revision, section_resource, nil)
   end
 
+  def get_combined_settings_for_all_resources(section_id, user_id) do
+    section = Oli.Delivery.Sections.get_section!(section_id)
+
+    all_pages = DeliveryResolver.all_pages(section.slug)
```

Comment:
`get_combined_settings_for_all_resources` seems like it is fetching a lot of information.  The call to `DeliveryResolver.all_pages/1` in a large course can return 500 or so full revisions, containing all JSON content.  What is actually needed here from the Revision? Only the title?  I think instead this call and perhaps the call to `get_section_resources` should be combined into new query that fetches the absolute minimum pieces of information needed.

Reply:
Reimplemented this query to be more efficient, returning the section resource and only the required settings